### PR TITLE
Metronome Release Process

### PR DESCRIPTION
**Pull request by kensipe**
Jenkin Release Process for Metronome

There are some small differences compared to Marathon.  Marathon version is tag based.  Metronome is based on checked in artifact.   

Jira:  MARATHON_EE-1725
https://jira.mesosphere.com/browse/MARATHON_EE-1725

### CODE REVIEW

**Comment by kensipe** [comment]
@timcharper I still can't add you as a reviewer but it would be awesome if you have time.

File: project/Build.scala [hunk 1]
```diff
@@ -82,7 +82,7 @@ object Build extends sbt.Build {
     organization := "dcos",
     scalaVersion := "2.11.8",
     crossScalaVersions := Seq(scalaVersion.value),
-    scalacOptions in Compile ++= Seq(
+    scalacOptions in (Compile, doc) ++= Seq(
```
Comment (alenkacz) [inline project/Build.scala]: what is the doc here?

File: ci/pipeline [hunk 2]
```diff
@@ -21,43 +21,43 @@ val PACKAGE_DIR: Path = pwd / 'target / 'universal
  * Compile Metronome and run unit and integration tests followed by scapegoat.
  */
 @main
-def compileAndTest(): Unit = utils.stage("Compile and Test") {
+def compileAndTest(logFileName: String): Unit = utils.stage("Compile and Test") {
```
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/pipeline]: scapegoat isn't part of this build... yet.    if we add it in, it should be part of a different pr.  there are a number issues last I looked.

File: project/Build.scala (hunk 1 above)
Comment (kensipe) [reply project/Build.scala]: this is the configuration necessary to added the option:  `"-no-link-warnings" // Suppresses problems with Scaladoc @throws links`

**Comment by kensipe** [comment]
@alenkacz all feedback addressed

File: project/plugins.sbt [hunk 3]
```diff
@@ -24,3 +24,5 @@ addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.0.4")
 addSbtPlugin("org.scoverage" % "sbt-coveralls" % "1.0.0")
 
 addSbtPlugin("com.github.gseitz" % "sbt-protobuf" % "0.5.2")
+
+addSbtPlugin("com.sksamuel.scapegoat" %% "sbt-scapegoat" % "1.0.4")
```
Comment (alenkacz) [inline project/plugins.sbt]: ok then I was just mislead by you adding a scapegoat plugin here - I thought you were planning to use it. So maybe we should remove this line and introduce scapegoat in a different PR?

Comment (kensipe) [reply project/plugins.sbt]: agree!

**Comment by kensipe** [comment]
@mesosphere-ci please retest this

File: ci/pipeline [hunk 4]
```diff
@@ -143,8 +143,46 @@ def run(): (String, Option[awsClient.Artifact]) = {
...
+    uploadTarballToS3(version, "builds")
   }
 }
+
+
+/**
+ * Executes the Community Release which includes:
+ *
+ *  1. tarball with version details 
+ *  Unlike the marathon build, the version details are currently expected to be
```
Comment (jeschkies) [inline ci/pipeline]: Please avoid the extra spaces after a punctuation. This makes the text a little less legible.

**Pull request merged**